src/main/java/Duke.java

```diff
@@ -1,10 +1,155 @@
+    private static final ArrayList<Task> list = new ArrayList<>();
+
+    private static void update(Task t) {
+        System.out.println("Got it. I've added this task:\n  "
+                + t.toString()
+                + "\nNow you have " + list.size() + " tasks in the list.");
+    }
+
+    private static void delete(int number) {
+        Task toDelete = list.get(number - 1);
+        list.remove(number - 1);
+        System.out.println("Noted. I've removed this task:\n  "
+                + toDelete.toString()
+                + "\nNow you have " + list.size() + " tasks in the list.");
+    }
+
+    private static void readAll() {
+        File f = new File("./Data/");
+        File f2 = new File("./Data/Duke.txt");
+        if (!f.exists()) {
+            f.mkdir();
+        }
+
+        try {
+            f2.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            BufferedReader b = new BufferedReader(new FileReader(f2));
+            String line = b.readLine();
+            while (line != null) {
+                String[] split = line.split("\\|");
+                switch(split[0]) {
+                case("T"):
+                    Todo t = new Todo(split[2]);
+                    if (!split[1].equals("0")) {
+                        t.markAsDone();
+                    }
+                    list.add(t);
+                    break;
+                case("E"):
+                    Event e = new Event(split[2], split[3]);
+                    if (!split[1].equals("0")) {
+                        e.markAsDone();
+                    }
+                    list.add(e);
+                    break;
+                case("D"):
+                    Deadline d = new Deadline(split[2], split[3]);
+                    if (!split[1].equals("0")) {
+                        d.markAsDone();
+                    }
+                    list.add(d);
+                    break;
+                }
+                line = b.readLine();
+            }
+            b.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void writeAll() {
+        try {
+            FileWriter w = new FileWriter("./Data/Duke.txt");
+            for (Task task : list) {
+                w.write(task.toWrite());
+            }
+            w.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo = "DUKE\n";
+        System.out.println("Hello from\n" + logo + "What can I do for you?");
+        readAll();
+        Scanner s = new Scanner(System.in);
```

Comment:
Variable `s` seems to have a large enough scope - it could have a longer, more descriptive name? I saw a similar naming for the `BufferedReader` variable  `b` above as well.